### ISSUE TITLE
PSR2 standard can't auto fix multi-line function call inside a string concat statement

### ISSUE THREAD

**Issue by stdex** [squizlabs/PHP_CodeSniffer#2506]
Hello.

PHP_CodeSniffer version 3.4.2 (stable) 

Call:  
phpcbf.bat --report=full -s -vv --standard=psr2  

Problem with code:  

```php
<?php

class C
{

    public function m()
    {
        $t = '
	' . (empty(true) ? '
	' . f(
                    '1',
                    '2',
                    '3',
                    '4'
                ) . '
	' : '');
    }

}
```

```
=> Fixing file: 4/7 violations remaining [made 50 passes]... 	* fixed 4 violations, starting loop 51 *
	*** Reached maximum number of loops with 4 violations left unfixed ***
ERROR in 1.62 secs

A TOTAL OF 3 ERRORS WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------
```

Problem with code:  
```php
<?php

class C
{

    public function m()
    {
        $a = [];
        $t =
            "SELECT * FROM t
WHERE f IN(" . implode(
                ",",
                $a
            ) . ")";
    }
}
```
```
=> Fixing file: 3/4 violations remaining [made 50 passes]... 	* fixed 3 violations, starting loop 51 *
	*** Reached maximum number of loops with 3 violations left unfixed ***
ERROR in 1.29 secs

A TOTAL OF 1 ERROR WERE FIXED IN 1 FILE
----------------------------------------------------------------------
PHPCBF FAILED TO FIX 1 FILE
----------------------------------------------------------------------
```

**Comment by jrfnl** [comment]
This is a fixer conflict between the `Generic.WhiteSpace.ScopeIndent` and the `PEAR.Functions.FunctionCallSignature` sniffs.

The problem (in both cases) is caused by the function call being concatenated onto a text string spanning multiple lines.

Looking at the code of the first example, IMO the correct fix would be to indent the code like below where the indent of the function call parameters is based on the indent of the `$t =` + 4 spaces, which would come down to 12 spaces.

Based on an initial check, it looks like both sniffs get it wrong.

The `PEAR.Functions.FunctionCallSignature` tries to fix to 4 spaces, i.e. start position of the line containing the function call (= 0 indent as it is the second line in a multi-line text string) + 4 spaces, while the `Generic.WhiteSpace.ScopeIndent` tries to set the indent to 8 spaces, i.e. the same as the start position of the `$t`.

```php
<?php

class C
{

    public function m()
    {
        $t = '
    ' . (empty(true) ? '
    ' . f(
            '1',
            '2',
            '3',
            '4'
        ) . '
    ' : '');
    }
}
```

**Comment by stdex** [comment]
Does this mean that I have to manually resolve such conflicts?

**Comment by jrfnl** [comment]
@stdex No, sorry, I wasn't clear. My post was just to add additional information to the issue in hopes that a solution can be found.

**Comment by gsherwood** [comment]
I've added a fix for the first case.

The PSR2 standard will now fix this code:
```php
<?php

class C
{

    public function m()
    {
        $t = '
        ' . (empty(true) ? '
        ' . f(
                        '1',
                        '2',
                        '3',
                        '4'
                    ) . '
        ' : '');
    }

}
```
So that it looks like this:
```php
<?php

class C
{

    public function m()
    {
        $t = '
        ' . (empty(true) ? '
        ' . f(
            '1',
            '2',
            '3',
            '4'
        ) . '
        ' : '');
    }
}
```

The second case is something slightly different and needs to be fixed separately.

**Comment by gsherwood** [comment]
I've added another fix into the same sniff for the second case.

The PSR2 standard will now fix this code:
```
<?php

class C
{

    public function m()
    {
        $a = [];
        $t =
            "SELECT * FROM t
WHERE f IN(" . implode(
                ",",
                $a
            ) . ")";
    }
}
```

So that it looks like this:
```
<?php

class C
{

    public function m()
    {
        $a = [];
        $t =
            "SELECT * FROM t
WHERE f IN(" . implode(
            ",",
            $a
        ) . ")";
    }
}
```
Thanks for the bug report.
